apps/web/src/components/settings/AccountSettings.tsx

```diff
@@ -5,33 +5,55 @@ import { Button } from "@/components/shadcn/ui/button";
 import { Label } from "@/components/shadcn/ui/label";
 import { toast } from "sonner";
 import { useState } from "react";
-import { users } from "db/schema";
-import { z } from "zod";
 import { useAction } from "next-safe-action/hook";
 import { modifyAccountSettings } from "@/actions/user-profile-mod";
+import { Checkbox } from "@/components/shadcn/ui/checkbox";
+import c from "config";
+import { Loader2 } from "lucide-react";
 
 interface UserProps {
 	firstName: string;
 	lastName: string;
+	email: string;
+	hackerTag: string;
+	hasSearchableProfile: boolean;
 }
+
 interface AccountSettingsProps {
 	user: UserProps;
 }
 
 export default function AccountSettings({ user }: AccountSettingsProps) {
 	const [newFirstName, setNewFirstName] = useState(user.firstName);
```

Comment:
non-blocking. When you have a lot of different `useState` hooks like this, it is better to use React Hook form as it handles a lot of the state changes, zod resolving, etc for you. The registration form uses them and also something good to note for later.